system/session/drivers/cookie.php

```diff
@@ -47,7 +48,7 @@ public function write($id, $cargo) {
 		$serialized = serialize($cargo);
 
 		// create a signature to verify content when unpacking
-		$sign = hash('md5', $serialized);
+		$sign = hash_hmac('sha256', $serialized, $session_key, true);
```

Comment:
I'm not an expert on this stuff, but I'm just wondering, if the stored password is in md5, wouldn't comparing an md5 hash and an sha256 hash return false always? We'd need to fix the stored passwords. Could we have another field in the users table of the database, called `created` which can hold the anchor version that the user was created on, and work from there to determine which algorithm to use, and whether we need to update their password (and hence the `created` field)?